packages/runtime/datastore/src/localChannelContext.ts

```diff
@@ -126,20 +119,17 @@ export abstract class LocalChannelContextBase implements IChannelContext {
 		trackState: boolean = false,
 		telemetryContext?: ITelemetryContext,
 	): Promise<ISummarizeResult> {
-		assert(
-			this.isLoaded && this.channel !== undefined,
-			0x18c /* "Channel should be loaded to summarize" */,
-		);
-		return summarizeChannelAsync(this.channel, fullTree, trackState, telemetryContext);
+		const channel = await this.getChannel();
+		return summarizeChannelAsync(channel, fullTree, trackState, telemetryContext);
 	}
 
 	public getAttachSummary(telemetryContext?: ITelemetryContext): ISummarizeResult {
 		assert(
-			this.isLoaded && this.channel !== undefined,
+			this.isLoaded && this._channel !== undefined,
```

Comment:
```suggestion
			this._channel !== undefined,
```

this was redundant before and now, looks like. might as well fix it to only use one (i lean toward _channel so below code typechecks)

Reply:
yeah. i was trying to minimize my changes, but isLoaded is redundant, as you need to type narrow anyway.